Mlem/Models/Content/Community/Community List/CommunityListModel.swift

```diff
@@ -205,28 +119,125 @@ class CommunityListModel: ObservableObject {
             )
             
             // as the call failed, we need to revert the change to the local state
-            await MainActor.run {
-                updateLocalStatus(for: community, subscribed: !subscribed)
-            }
+            await updateLocalStatus(for: community, subscribed: !subscribed)
         }
     }
     
-    private func updateFavorites(_ favorites: [APICommunity]) {
-        combine(subscriptions, favorites)
+    private func updateFavorites(_ favorites: [APICommunity]) async {
+        await combine(subscribed, favorites)
     }
     
-    private func combine(_ subscriptions: [APICommunity], _ favorites: [APICommunity]) {
-        // store the values for future use...
-        self.subscriptions = subscriptions
-        favoriteCommunities = favorites
+    private func combine(_ subscribed: [APICommunity], _ favorited: [APICommunity]) async {
+        // store the values for future use
+        self.subscribed = subscribed
+        subscribedSet = Set(subscribed.map(\.id))
+        self.favorited = favorited
+        favoritedSet = Set(favorited.map(\.id))
         
         // combine and sort the two lists, excluding duplicates
-        let combined = subscriptions + favorites.filter { !subscriptions.contains($0) }
+        let combined = subscribed + favorited.filter { !subscribed.contains($0) }
         let sorted = combined.sorted()
+  
+        // Task {
+        await MainActor.run {
+            self.allCommunities = sorted
+        }
+        let (newAllSections, newVisibleSections) = recomputeSections()
+        await MainActor.run {
+            self.allSections = newAllSections
+            self.visibleSections = newVisibleSections
+        }
+        // }
+    }
+    
+    // swiftlint:disable:next function_body_length
+    private func recomputeSections() -> (all: [CommunityListSection], visible: [CommunityListSection]) {
+        var newAllSections: [CommunityListSection] = .init()
+        var newVisibleSections: [CommunityListSection] = .init()
+        
+        let topSection = withDependencies(from: self) {
+            CommunityListSection(
+                viewId: "top",
+                sidebarEntry: EmptySidebarEntry(
+                    sidebarLabel: nil,
+                    sidebarIcon: "line.3.horizontal"
+                ),
+                inlineHeaderLabel: nil,
+                accessibilityLabel: "Top of communities",
+                communities: .init()
+            )
+        }
+        newAllSections.append(topSection)
+        
+        let favoritesSection = withDependencies(from: self) {
+            CommunityListSection(
+                viewId: "favorites",
+                sidebarEntry: FavoritesSidebarEntry(
+                    sidebarLabel: nil,
```

Comment:
I noticed that jumping to the favorites section doesn't work correctly, though it doesn't currently work on `dev` either so not a problem with this PR

Reply:
Opened a separate issue for it